pytorch_blade/src/compiler/mlir/converters/impl/shapes.cpp

```diff
@@ -287,6 +287,82 @@ bool ConvertAtenSize(MhloConversionContext& ctx, const torch::jit::Node& node) {
   return true;
 }
 
+bool ConvertAtenRoll(MhloConversionContext& ctx, const torch::jit::Node& node) {
+  const auto& loc = GetNodeLocation(ctx, node);
+  auto jit_self = node.input(0);
+  auto jit_shifts = node.input(1);
+  auto jit_dims = node.input(2);
+  if (!CheckConstAttribute(jit_shifts, "aten::roll", "shifts")) {
+    return false;
+  }
+  if (!CheckConstAttribute(jit_dims, "aten::roll", "dims")) {
+    return false;
+  }
+
+  auto shifts_ival = torch::jit::toIValue(jit_shifts);
+  if (!(shifts_ival && shifts_ival->isIntList())) {
+    DLOG(WARNING) << "aten::roll shifts must be constants";
+    return false;
+  }
+
+  auto dims_ival = torch::jit::toIValue(jit_dims);
+  if (!(dims_ival && dims_ival->isIntList())) {
+    DLOG(WARNING) << "aten::roll dims must be constants";
+    return false;
+  }
+  auto shifts = shifts_ival->toIntList();
+  auto dims = dims_ival->toIntList();
+
+  mlir::Value self = ctx.GetMlirValue(jit_self);
+  auto builder = *ctx.builder;
+  for (size_t d = 0; d < dims.size(); ++d) {
+    self = mlir::mhlo::BuildRoll(builder, loc, self, shifts[d], dims[d]);
+  }
+  ctx.value_map[node.output()] = self;
+
+  return true;
+}
+bool ConvertAtenUnbind(
+    MhloConversionContext& ctx,
+    const torch::jit::Node& node) {
+  if (!GetTrustTracingShape()) {
+    // Doesn't support dynamic outputs, and our shape are recorded by
+    // tracing. We won't do conversion if we don't trust the tracing shape.
+    //
+    // TODO(gty):
+    // This could be removed once we have more trustworthy shapes,
+    // such as those analyzed from static shape analysis.
+    return false;
+  }
+
+  const auto& loc = GetNodeLocation(ctx, node);
+  auto jit_self = node.input(0);
+  auto jit_dim = node.input(1);
+
+  if (!CheckConstAttribute(jit_dim, "aten::unbind", "dim")) {
+    return false;
+  }
+  mlir_dim_t dim = CastJitConstToInt64(*jit_dim);
+
+  auto tensor_type = jit_self->type()->cast<torch::TensorType>();
+  TORCH_CHECK(tensor_type != nullptr);
+
+  // NB: we get number of outputs from the Tensor's shape.
+  auto tensor_sizes = *(tensor_type->sizes().concrete_sizes());
+  auto dim_size = tensor_sizes[dim];
+  auto self = ctx.GetMlirValue(jit_self);
+  auto builder = *ctx.builder;
```

Comment:
nit: maybe we don't have to create a copy of the builder? `BuildStdConstForI64` and `BuildSelect` just receive references.